world-server/src/world/Math.ts

```diff
@@ -2,3 +2,11 @@ export interface Vector2 {
     x: number;
     y: number;
 }
+
+export function vectorAdd(vec1: Vector2, vec2: Vector2): Vector2 {
```

Comment:
Would suggest using as static functions inside Vector2 if possible (less imports, quasi namespaced). This is if it's possible to convert to abstract class without breaking changes.

Reply:
The Vector2 interface is used in message handling and it is an interface precisely because the client sends simple objects with no type.